### ISSUE TITLE
代码问题

### ISSUE THREAD

**Issue by LilySys** [michuanhaohao/reid-strong-baseline#65]
罗博士，您好！
       我有两个问题希望您能释疑：
    （一）reid_metric.py中的distmat = torch.pow(qf, 2).sum(dim=1, keepdim=True).expand(m, n) + torch.pow(gf, 2).sum(dim=1, keepdim=True).expand(n, m).t()中的torch.pow(qf, 2).sum(dim=1, keepdim=True)为什么求和之后的值都为1？
     （二）reid_metric.py中的distmat.addmm_(1, -2, qf, gf.t())中的参数为什么设置为1，-2？

**Comment by michuanhaohao** [comment]
（1）说明你对特征向量做了L2归一化，所以模长都为1
（2）这句话的意思就是distmat-2*qf*gf，而distmat=qf^2+gf^2，这是L2距离的张量写法，需要自己体会一下

**Comment by LilySys** [comment]
好的，非常感谢！

**Comment by LilySys** [comment]
罗博士，您好！
       接着上面您的回答，feats = torch.nn.functional.normalize(feats, dim=1, p=2)已做归一化，tensor中每一行的元素直接相加已经等于1，为什么还要torch.pow(qf, 2).sum(dim=1, keepdim=True)？这个是不是有点多余？

**Comment by michuanhaohao** [comment]
@LilySys 因为测试的时候有可能不做归一化，那么这句就很重要了，为了统一写法保留了
